libs/core/src/lib/wizard/wizard.component.ts

```diff
@@ -0,0 +1,211 @@
+    }
+
+    /** @hidden */
+    private _setContentTemplate(): void {
+        this.steps.forEach((step) => {
+            step.finalStep = false;
+            if (step.status === 'current' && step.content) {
+                this.contentTemplate = step.content.contentTemplate;
+            }
+        });
+        this.steps.last.finalStep = true;
+    }
+
+    /** @hidden */
+    private _hideSomeStep(currentStep: WizardStepComponent): void {
+        // If a small step was found, get the step with a visible label furthest away from the current step and hide the label
+        let stepsArray = this.steps.toArray();
+        stepsArray = stepsArray.filter((step) => {
+            return !step.elRef.nativeElement.classList.contains(STEP_NO_LABEL_CLASS);
+        });
+        if (stepsArray.length > 1) {
+            let currentStepIndex = 0,
+                stepToHide;
+            if (currentStep) {
+                currentStepIndex = stepsArray.indexOf(currentStep);
+            }
+            currentStepIndex > (stepsArray.length - 1) / 2
+                ? (stepToHide = stepsArray[0])
+                : (stepToHide = stepsArray[stepsArray.length - 1]);
+            stepToHide.elRef.nativeElement.classList.add(STEP_NO_LABEL_CLASS);
+            stepToHide.elRef.nativeElement.classList.add(STEP_STACKED_CLASS);
+            this._setStackedTop(currentStep);
+        }
+    }
+
+    /** @hidden */
+    private _setStackedTop(currentStep: WizardStepComponent): void {
+        this.steps.forEach((step, index) => {
+            step.elRef.nativeElement.classList.remove(STEP_STACKED_TOP_CLASS);
+            if (this.steps.toArray()[index + 1] === currentStep) {
+                step.elRef.nativeElement.classList.add(STEP_STACKED_TOP_CLASS);
+            }
+        });
+    }
+
+    /** @hidden */
+    private _resetStepClasses(): void {
+        this.steps.forEach((step) => {
+            step.elRef.nativeElement.classList.remove(STEP_STACKED_TOP_CLASS);
+            step.elRef.nativeElement.classList.remove(STEP_STACKED_CLASS);
+            step.elRef.nativeElement.classList.remove(STEP_NO_LABEL_CLASS);
+        });
+    }
+
+    /** @hidden */
+    private _handleStepOrStatusChanges(): void {
+        this._setContentTemplate();
+        this._shrinkWhileAnyStepIsTooNarrow();
+        this._cdRef.detectChanges();
+    }
+
+    /** @hidden */
+    private _shrinkWhileAnyStepIsTooNarrow(): void {
+        this._resetStepClasses();
+        let i = 0;
+        while (this._anyStepIsTooNarrow() && i < this.steps.length - 1) {
+            i++;
+            this._wizardShrinking();
+        }
+    }
+
+    /** @hidden */
+    private _anyStepIsTooNarrow(): boolean {
+        let foundNarrowStep = false;
+        this.steps.forEach((step) => {
+            if (step.elRef.nativeElement.clientWidth < STEP_MIN_WIDTH) {
+                foundNarrowStep = true;
+            }
+        });
+        return foundNarrowStep;
```

Comment:
`Array.some` will give less code
`const foundNarrowStep = this.steps.some(step => step.elRef.nativeElement.clientWidth < STEP_MIN_WIDTH);`